packages/purpose-process/src/model/domain/models.ts

```diff
@@ -22,3 +27,12 @@ export type ApiRiskAnalysisFormSeed = z.infer<
 export type ApiReversePurposeUpdateContent = z.infer<
   typeof api.schemas.ReversePurposeUpdateContent
 >;
+
+export type ApiGetPurposesFilters = {
+  name?: string;
+  eservicesIds: EServiceId[];
+  consumersIds: TenantId[];
+  producersIds: TenantId[];
+  states: PurposeVersionState[];
+  excludeDraft: boolean | undefined;
+};
```

Comment:
This type could be derived from zodios

```suggestion
export type ApiGetPurposesFilters = ZodiosQueryParamsByPath<
  typeof api.api.api,
  "get",
  "/purposes"
>;
```

The `ZodiosQueryParamsByPath` type utility must be imported from `@zodios/core`

Reply:
I removed that type, but I still need a custom type for the query because the one from the api spec is not parsed to what we need.

Reply:
May I ask you what the differences were? 
The `Api-` types should be equal to what the actual API accepts/sends

Reply:
I got it, but we need to [parse](https://github.com/pagopa/interop-be-monorepo/pull/438/files#diff-e84c431f3ccf88a6672169ab51526ca77a391d4c82e91d751f4a3263c65c9423R96-R100) what we get in input.

Reply:
As we discussed in the call, it's okkay for me to not derive the type.
The typing from zodios is not compatible with the one we use in the service. 
Working into making it compatible is not worth the effort.